internal/handshake/xor_nonce_aead_boring.go

```diff
@@ -41,23 +39,21 @@ func (f *xorNonceAEAD) sealZeroNonce() {
 }
 
 func (f *xorNonceAEAD) seal(nonce, out, plaintext, additionalData []byte) []byte {
-	if useBoring {
-		if boring.Enabled() {
-			if !f.hasSeenNonceZero {
-				// BoringSSL expects that the first nonce passed to the
-				// AEAD instance is zero.
-				// At this point the nonce argument is either zero or
-				// an artificial one will be passed to the AEAD through
-				// [sealZeroNonce]
-				f.hasSeenNonceZero = true
-				if !allZeros(nonce) {
-					f.sealZeroNonce()
-				}
-			}
-		} else {
+	if !goBoringDisabled {
+		if !boring.Enabled() {
```

Comment:
this if is impossible to happen right? because of the build feature.

Reply:
No, you can use a vanilla Go compiler and set the buildtags to include the boringcrypto which will compile this source file however at runtime it will crash.

Reply:
Can we add a comment to this [line](https://github.com/chungthuang/quic-go/pull/7/files#diff-79028f8170c54b5da11d1c1ffa60584821f289f59a5fd5e40f6a77dfa9697e53R15), or in the README because we are changing the behavior of this code to require to be built with goboring or you need to tell the compiler explicitly that it's not required.